src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/IProvenBlockHeaderStore.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Cache layer for <see cref="ProvenBlockHeader"/>s.
+    /// </summary>
+    public interface IProvenBlockHeaderStore : IProvenBlockHeaderProvider, IDisposable
+    {
+        /// <summary>
+        /// Adds a <see cref="ProvenBlockHeader"/> to pending batch concurrent dictionary.
```

Comment:
adds to pending batch (items that will be saved) and cache.
Also don't mention concurrent dictionary because it's an implementation detail.

Reply:
ok